pkg/apis/experimental/types.go

```diff
@@ -639,3 +639,36 @@ type ClusterAutoscalerList struct {
 
 	Items []ClusterAutoscaler `json:"items"`
 }
+
+// Selector is a label query over a set of API objects. MatchLabels and
+// MatchExpressions are ANDed. An empty Selector matches all objects. A
+// null Selector matches no objects.
+type Selector struct {
+	// MatchLabels is a list of labels.
+	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+	// MatchExpressions is a list of requirement selectors.
```

Comment:
Dug up another argument against OR from old notes:

Some amount of homogeneity is expected in sets targeted by services and managed by controllers. Operators feel it's better to explicitly express and require such homogeneity, such as by common labels/keys, and to make accidental heterogeneity of totally unrelated things hard.